test.mjs

```diff
@@ -12,20 +13,26 @@ testContext(esm, 'esm');
 
 function testContext({any, html, css, gql, md}, name) {
 	describe(name + ' imports', () => {
-		it('base string', () => {
+		it('exports', () => {
+			assert.equal(any, html);
+			assert.equal(any, css);
+			assert.equal(any, gql);
+			assert.equal(any, md);
+		});
+
+		it('code-tags', () => {
 			assert.equal(any`a`, 'a');
-			assert.equal(html`a`, 'a');
-			assert.equal(css`a`, 'a');
-			assert.equal(gql`a`, 'a');
-			assert.equal(md`a`, 'a');
+			assert.equal(any` a `, ' a ', 'Preserve boundary whitespace');
+			assert.equal(any`a${'b'}c${1}`, 'abc1', 'Interpolate with strings and numbers');
+			assert.equal(any`\\\na${'\\\na'}`, '\\\na\\\na', 'Preserve escape sequences');
+			assert.equal(any`🇪🇺 ${'🇺🇳'}`, '🇪🇺 🇺🇳', 'Preserve combined emojis');
 		});
 
-		it('template with interpolation', () => {
-			assert.equal(any`a${'b'}c${1}`, 'abc1');
-			assert.equal(html`a${'b'}c${1}`, 'abc1');
-			assert.equal(css`a${'b'}c${1}`, 'abc1');
-			assert.equal(gql`a${'b'}c${1}`, 'abc1');
-			assert.equal(md`a${'b'}c${1}`, 'abc1');
+		it('stringifiable objects', () => {
+			const stringifiableObject = new Date();
+			const stringifiedObject = String(stringifiableObject);
+
+			assert.equal(any`${stringifiableObject}`, stringifiedObject, 'Interpolate objects with toString');
```

Comment:
```suggestion
			const date = new Date();
			const dateString = String(date);

			assert.equal(any`${date}`, dateString, 'Interpolate objects with toString');

```

The names are not so good, is there anything in js can't "stringify"?

Reply:
I just wanted to make it clear that I don’t really care it’s a date. My first attempt was `{toString: () => 'a'}`

Reply:
> The names are not so good, is there anything in js can't "stringify"?

```js
String(Object.create(null))
// TypeError: No default value
```